lms/djangoapps/course_blocks/api.py

```diff
@@ -91,6 +93,8 @@ def get_course_blocks(
     """
     if not transformers:
         transformers = BlockStructureTransformers(get_course_block_access_transformers(user))
+    if include_completion:
+        transformers += [BlockCompletionTransformer()]
```

Comment:
The reason I ended up implementing it this way is because when you specify transformers to pass in, then you don't get the transformers from `get_course_block_access_transformers` automatically. After looking into those, they mostly all seem like transformers we want to keep (hiding additional content library problems, content type gating, taking user partitions into account in case there is a subsection hidden to only one enrollment track, visibility for blocks marked staff only, and then the date transformer from edx_when). It looks like the only one I would feel comfortable removing would be the `StartDateTransformer`, but I am bypassing that with the `allow_start_dates_in_future` flag. So in the end, I thought it was cleaner to include this flag rather than define most. Let me know if you think it would be better to just specify all the blocks.

Reply:
My main concern is whether `include_completion` is basically a single-use flag (and not generally applicable), or whetehr we expect `include_completion` to be used in a bunch of places. If it's single-use, then I'd prefer to duplicate the default transformers out into the caller, or amend this function to allow you to specify both default and additional transformers, or something.

Reply:
I guess I'm not really sure how people want to use it. To me, I would think knowing block completion is something others could want which is why I opted for adding it in there for others to use. I did think about changing the behavior to make the transformers in `get_course_block_access_transformers` added in regardless, but I didn't want to change the behavior of the function for everyone which is why I opted for the additional flag. Happy to pull it out if you think that would be better. Easy switch to just have 
```
transformers = get_course_block_access_transformers(user)
transformers += [BlockCompletionTransformer()]
block_data = get_course_blocks(user, course_usage_key, transformers, allow_start_dates_in_future=True, include_completion=True)

Reply:
@cpennington I'm going to opt for leaving it the way it is. It was something like this that let me know the Block Completion Transformer even existed in the first place and I think as we encourage more usage of the course_blocks_api, it will help people know which transformers exist and how they can add in a transformer. If we decide at some point to instead make all of the access checks non-optional, then it will be easy to pull it out (by just searching for wherever `include_completion` is defined) and go with the defining the transformers route, but that could be changing behavior for people so I want to be careful before making that decision.